libbeat/cmd/instance/metrics.go

```diff
@@ -55,13 +59,29 @@ func setupMetrics(name string) error {
 		IncludeTop:   process.IncludeTopConfig{},
 	}
 	err := beatProcessStats.Init()
+	if err != nil {
+		return err
+	}
+
+	beatProcessSysInfo, err = sysinfo.Self()
+	if err != nil {
+		return err
+	}
 
-	return err
+	var ok bool
+	handleCounter, ok = beatProcessSysInfo.(types.OpenHandleCounter)
```

Comment:
Not all platforms implement `types.OpenHandleCounter` so if it's unavailable then don't report the number of open handles. At current time we don't have an implementation for darwin in go-sysinfo, but you could fallback to go-sigar in the meantime.